gcc/rust/ast/rust-stmt.h

```diff
@@ -204,28 +173,21 @@ class ExprStmtWithoutBlock : public ExprStmt
 class ExprStmtWithBlock : public ExprStmt
 {
 public:
-  // ExprWithBlock* expr;
-  ::std::unique_ptr<ExprWithBlock> expr;
-
-  /*~ExpressionStatementWithBlock() {
-      delete expr;
-  }*/
+  std::unique_ptr<ExprWithBlock> expr;
```

Comment:
I know i have added these as public fields for the passes which seems kind of messy but i cant think of anything better right now. I think the answer will become clear over time i think we should raise an issue about them.

Reply:
I agree. Other ASTs using this inheritance model (e.g. clang) seem to have getters and setters, though they also don't seem to manage memory in the same way.